### PR TITLE
fix: Float to string serialization improvements with better accuracy.

### PR DESCRIPTION
**Pull request by amunra**
Now using the [`dtoa.c` library](http://www.netlib.org/fp/) to convert `double` values to strings.
The library encodes floating points as strings that should round-trip back to doubles accurately.

This change fixes issues #3 (Question/Bug: Support more than 6 significant digits).

### CODE REVIEW

File: src/mem_writer.c [hunk 1]
```diff
@@ -139,8 +141,106 @@ void mem_writer_i64(mem_writer* writer, int64_t num)
...
+    char* buf_end = NULL;
+
+    // Returns buffer of integer digits or special value, no '.' chars.
+    char* buf = dtoa(num, mode, ndigits, &decpt, &sign, &buf_end);
+    assert((buf_end != NULL) && (buf_end >= buf));
+
+    const size_t len = buf_end - buf;
+    assert(len != 0);
+
+    if (buf[0] == 'N')
```
Comment (eugenels) [inline src/mem_writer.c:169]: NaN/Inf values can be tested with isnan/isinf direcly

File: src/mem_writer.c [hunk 2]
```diff
@@ -139,8 +141,106 @@ void mem_writer_i64(mem_writer* writer, int64_t num)
...
+            mem_writer_str(writer, 9, "-Infinity");
+        else
+            mem_writer_str(writer, 8, "Infinity");
+    }
+    else if (decpt <= 0)
+    {
+        char* dest = NULL;
+        const size_t leading_zeros = (size_t)(-decpt);
+        size_t book_len = 2 + leading_zeros + len;
+        if (sign)
```
Comment (eugenels) [inline src/mem_writer.c:185]: Can be converted to
```
        size_t book_len = sign + 2 + leading_zeros + len;
        dest = mem_writer_book(writer, book_len);
        if (sign) *dest++ = '-';
```
and save a few lines of code.

Comment (amunra) [reply src/mem_writer.c:185]: Good point, almost. Really should be `size_t book_len = !!sign + 2 + leading_zeros + len;` (as sign may not be exatcly 1).

Comment (eugenels) [reply src/mem_writer.c:185]: You are right.
Anyway dtoa_r set it to 1 as was expected.

**Comment by amunra** [comment]
Abandoning change. Will use a different solution that causes fewer threading constraints.

**Pull request closed without merging**